pkg/pipeline/encode/encode_prom_test.go

```diff
@@ -62,124 +66,238 @@ parameters:
             type: histogram
             valueKey: aggregate
             labels:
+          - name: subnetHistogram2
+            type: agg_histogram
+            valueKey: aggregate
+            labels:
 `
 
-func initNewEncodeProm(t *testing.T) Encoder {
-	v, cfg := test.InitConfig(t, testConfig)
-	require.NotNil(t, v)
-
-	newEncode, err := NewEncodeProm(cfg.Parameters[0])
-	require.Equal(t, err, nil)
-	return newEncode
+func initPromWithServer(params *api.PromEncode) (*EncodeProm, func(), error) {
+	reg := prometheus.NewRegistry()
+	prometheus.DefaultRegisterer = reg
+	prometheus.DefaultGatherer = reg
+	http.DefaultServeMux = http.NewServeMux()
```

Comment:
What's the purpose of this setting?

Reply:
this is because in `encode_prom`, we run `http.Handle("/metrics", promhttp.Handler())`
which creates a handler on the default mux/router. When this is called several times, an error would be fired (something like "cannot register /metrics, route already exists").

I don't remember exactly in which case that happened, maybe it was just in the benchmark I created below.

BTW it also shows that the prom_encode stage will need to be refactored if I some point we want to be able to define more than one prom-encodes in a pipeline. I mentioned that in this jira: https://issues.redhat.com/browse/NETOBSERV-498

Reply:
@jotak thanks. I tried commenting out that line and see what happens. It failed on the second unit test of `encode_prom_test.go` because of the multiple registration that you have described. So resetting the `DefaultServeMux` to a new instance on each unit test solves this problem.